### ISSUE TITLE
Weird connect equations present in MSL3.2 and some user models

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#642]
**Reported by adrpo on 20 Nov 2011 08:51 UTC**

Hi all,

There are several connect equations that connect the same
connector instance, i.e. connect(A, A) present in MSL 3.2
(we also found some of these in some user models).

Question:
Is there some voodoo going on or these are just no-ops?
There's nothing in the specification about something like this.
Currently we just issue a warning and ignore them.

For example:
```
[Modelica 3.2/Electrical/QuasiStationary/MultiPhase.mo:1872:7-1873:76:writable] Warning: connect(currentP, currentP) connects the same connector instance! The connect equation will be ignored.
[Modelica 3.2/Electrical/Digital.mo:5309:11-5312:35:writable] Warning: connect(dataOut, dataOut) connects the same connector instance! The connect equation will be ignored.
[Modelica 3.2/Electrical/Digital.mo:6206:11-6209:35:writable] Warning: connect(dataOut, dataOut) connects the same connector instance! The connect equation will be ignored.
[Modelica 3.2/Electrical/Digital.mo:6678:11-6681:35:writable] Warning: connect(dataOut, dataOut) connects the same connector instance! The connect equation will be ignored.
[Modelica 3.2/Electrical/Machines.mo:3880:9-3883:33:writable] Warning: connect(plug_rn, plug_rn) connects the same connector instance! The connect equation will be ignored.
```

Cheers,
Adrian Pop/

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/642

**Comment by modelica-trac-importer** [comment]
**Comment by pharman on 20 Nov 2011 10:41 UTC**
See also #552, though that doesn't answer the question of why they are there to start with.

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 20 Nov 2011 11:05 UTC**

Thanks for the pointer Peter.
So there is some voodoo going on after all :)
Yet another classic special case that we're always glad to discover.
Well, at least now I know how we should handle them.

Cheers,
Adrian Pop/

**Comment by modelica-trac-importer** [comment]
**Comment by volker&#46;beuter on 20 Nov 2011 23:29 UTC**
This is not a new problem. In a project working with customer libraries we are struggling
with lots of those self connections, mostly on data busses. No one says he had done such
connections purposely. These libraries have been created with Dymola version 6.1 and earlier
so I suspect it must have been easy to create self connections accidently with these versions
and there have been such self connections also in earlier MSL versions.
Tools ought to give at least a warning in case such a connections has been created
during graphical composition of a model. For example Dymola 2012 does not give any warning
on this - but here I had to create it on purpose. (This is of course no matter of the
specification.)

**Comment by modelica-trac-importer** [comment]
**Comment by majetta on 21 Nov 2011 09:00 UTC**
The connections in

```
[Modelica 3.2/Electrical/Digital.mo:5309:11-5312:35]
[Modelica 3.2/Electrical/Digital.mo:6206:11-6209:35] 
[Modelica 3.2/Electrical/Digital.mo:6678:11-6681:35]
```
are removed in c60c4ebb38874c004c9a4a68705c0a0e7eead8fb.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 21 Nov 2011 09:39 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 22 Nov 2011 09:08 UTC**
Could you please fix MSL3.2 maintenance branch, too!

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 22 Nov 2011 12:58 UTC**
Fixed for `Machines.mo` and `MultiPhase.mo` in 7a502aa4284d83f906896047c79c211ee9fd8ed4

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 22 Nov 2011 13:39 UTC**
Fixed in 84f7ae8892d294218bcbbb53552d09ebc216addb : 6d9780ab59ba8d6dbcd1cb212c93661ab7390711 for 3.2 maintenance.
